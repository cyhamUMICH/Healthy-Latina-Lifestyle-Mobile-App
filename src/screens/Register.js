import React from 'react';
import { View, Text } from 'react-native';
import { styles } from '../styles/Styles';
import { Button } from 'react-native-elements';
import { TextInput } from 'react-native-gesture-handler';

const Register = (props) => {
  return (
    <View style={styles.app}>

      <View style={styles.inputView}>
        <TextInput
          style={styles.TextInput}
          placeholder="name"
          placeholderTextColor="#003f5c"
          //onChangeText={(email) => setEmail(email)}
        />
      </View>
 
      <View style={styles.inputView}>
        <TextInput
          style={styles.TextInput}
          placeholder="email"
          placeholderTextColor="#003f5c"
          //onChangeText={(password) => setPassword(password)}
        />
      </View>

      <View style={styles.inputView}>
        <TextInput
          style={styles.TextInput}
          placeholder="username"
          placeholderTextColor="#003f5c"
          //onChangeText={(password) => setPassword(password)}
        />
      </View>

      <View style={styles.inputView}>
        <TextInput
          style={styles.TextInput}
          placeholder="password"
          placeholderTextColor="#003f5c"
          secureTextEntry={true}
          //onChangeText={(password) => setPassword(password)}
        />
      </View>

      <Button 
        buttonStyle={styles.basicButtons}
        title="Create Account"  onPress={() => props.navigation.navigate("Login")} 
        />


    </View>
  );
};

export default Register;