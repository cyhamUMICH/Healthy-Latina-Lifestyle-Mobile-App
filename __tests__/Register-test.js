// Code from: https://github.com/byCedric/expo-guide-ci/blob/master/__tests__/App-test.js
//      and https://jestjs.io/docs/tutorial-react-native
import React from 'react';
import Register from '../src/screens/Register';
import renderer from 'react-test-renderer';

test('Renders Register With the Loading Screen', async () => {
  const tree = renderer.create(<Register />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders Register Without the Loading Screen', async () => {
  const tree = renderer.create(<Register skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});