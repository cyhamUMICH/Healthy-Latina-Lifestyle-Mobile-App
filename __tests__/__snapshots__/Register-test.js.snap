// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Register With the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": "2%",
        "paddingTop": "2%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "padding": "10%",
        }
      }
    >
      <View
        style={
          Object {
            "marginVertical": "2%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="name"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="email"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="username"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="password"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00A89D",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": "2%",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Create Account
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View />
  </View>
</View>
`;

exports[`Renders Register Without the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": "2%",
        "paddingTop": "2%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "padding": "10%",
        }
      }
    >
      <View
        style={
          Object {
            "marginVertical": "2%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="name"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="email"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="username"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="password"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00A89D",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": "2%",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Create Account
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View />
  </View>
</View>
`;
